batchflow/models/tf/base.py

```diff
@@ -242,76 +245,153 @@ def __init__(self, *args, **kwargs):
         self.session = kwargs.get('session', None)
         self.graph = tf.Graph() if self.session is None else self.session.graph
         self._graph_context = None
-        self.is_training = None
-        self.global_step = None
-        self.loss = None
-        self.microbatch = None
-        self.train_steps = None
-        self.optimizers = {}
-        self._train_lock = threading.Lock()
         self._full_config = {}
-        self._attrs = []
         self._saver = None
-        self._to_classes = {}
-        self._inputs = {}
-        self.inputs = None
 
-        super().__init__(*args, **kwargs)
+        # Train configurations
+        self.train_steps = None
+        self._train_lock = threading.Lock()
 
-    def build(self, *args, **kwargs):
-        """ Build the model
-
-        #. Create a graph
-        #. Define ``is_training`` and ``global_step`` tensors
-        #. Define a model architecture by calling :meth:``._build``
-        #. Create a loss function from config
-        #. Create an optimizer and define a train step from config
-        #. `Set UPDATE_OPS control dependency on train step
-           <https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization>`_
-        #. Create a tensorflow session
-        """
+        # Parameters of batch processing: splitting batches into parts and/or using multiple devices to process data
+        self.microbatch = None
+        self.devices = []
+        self.leading_device = None
+        self.device_to_scope = {}
+        self.multi_device = False
+
+        # Private storage for often used tensors
+        self._attrs = {}
+
+        super().__init__(*args, **kwargs)
 
-        def _device_context():
-            if 'device' in self.config:
-                device = self.config.get('device')
-                context = self.graph.device(device)
+    def store_to_attr(self, attr, graph_item, device=None):
+        """ Store `graph_item` to private container."""
+        if self._attrs.get(attr) is None:
+            if device is None:
+                self._attrs[attr] = graph_item
+            else:
+                self._attrs[attr] = {device: graph_item}
+        else:
+            if device is None:
+                self._attrs[attr] = graph_item
             else:
-                context = contextlib.ExitStack()
-            return context
+                self._attrs[attr][device] = graph_item
+
+    def get_from_attr(self, attr, device=None, default=None):
+        """ Get item from private container or directly from model graph."""
+        device = device or self.leading_device
+        if attr in self._attrs:
+            if isinstance(self._attrs[attr], dict):
+                if device in self._attrs[attr]:
+                    graph_item = self._attrs[attr][device]
```

Comment:
```suggestion
                    return self._attrs[attr][device]
```
It's `python` so you can write a shorter code